src/modules/SX126x/SX126x.cpp

```diff
@@ -425,12 +435,23 @@ int16_t SX126x::packetMode() {
 }
 
 int16_t SX126x::scanChannel() {
-  return(this->scanChannel(RADIOLIB_SX126X_CAD_PARAM_DEFAULT, RADIOLIB_SX126X_CAD_PARAM_DEFAULT, RADIOLIB_SX126X_CAD_PARAM_DEFAULT));
-}
-
-int16_t SX126x::scanChannel(uint8_t symbolNum, uint8_t detPeak, uint8_t detMin) {
+  ChannelScanConfig_t config = {
+    .cad = {
+      .symNum = RADIOLIB_SX126X_CAD_PARAM_DEFAULT,
+      .detPeak = RADIOLIB_SX126X_CAD_PARAM_DEFAULT,
+      .detMin = RADIOLIB_SX126X_CAD_PARAM_DEFAULT,
+      .exitMode = RADIOLIB_SX126X_CAD_PARAM_DEFAULT,
+      .timeout = 0,
+      .irqFlags = RADIOLIB_IRQ_NOT_SUPPORTED,
```

Comment:
This should be `RADIOLIB_IRQ_CAD_DEFAULT_FLAGS`. When calling `scanChannel()` without arguments, it was waiting forever in my testing. The config in `startChannelScan()` has it correct.

Reply:
Good catch - thank you! I probably forgot to change this at some point. Tehre was also a similar issue in the LR11x0 driver.